k2/csrc/array.h

```diff
@@ -161,30 +161,29 @@ class Array1 {
     return ans;
   }
 
-  // Resizes, copying old contents if we could not re-use the same memory
-  // location. It will always at least double the allocated size if it has to
-  // reallocate. See Region::num_bytes vs. Region::bytes_used.
-  // TODO(haowen): now we only support the case that the current array `curr`
-  // (i.e. the array that will be resized) covers the highest used index in
-  // the region, that is, for any array `a` uses this region,
-  // curr.byte_offset_ + curr.Dim() * curr.ElementSize() == region_->bytes_used
-  // >= a.byte_offset + a.Dim() * a.ElementSize()
+  /*
+   Extend size of array, copying old contents if we could not re-use the same memory
```

Comment:
Maybe it's better to update documentation here as we support `reduce array size` in `Resize` here? As I just feel `Extend` means that making it longer or doing nothing (as you did in Region.Extend())

Reply:
OK, will do.